Core/include/Acts/TrackFitting/GaussianSumFitter.hpp

```diff
@@ -349,47 +350,35 @@ struct GaussianSumFitter {
 
       using PM = TrackStatePropMask;
 
-      typename propagator_t::template action_list_t_result_t<
-          MultiComponentBoundTrackParameters,
-          decltype(bwdPropOptions.actionList)>
-          inputResult;
-
-      // Unfortunately we must construct the result type here to be able to
-      // return an error code
-      using ResultType =
-          decltype(m_propagator.template propagate<
-                   MultiComponentBoundTrackParameters, decltype(bwdPropOptions),
-                   MultiStepperSurfaceReached>(
-              std::declval<MultiComponentBoundTrackParameters>(),
-              std::declval<Acts::Surface&>(),
-              std::declval<decltype(bwdPropOptions)>(),
-              std::declval<decltype(inputResult)>()));
-
-      auto& r = inputResult.template get<typename GsfActor::result_type>();
-
-      r.fittedStates = &trackContainer.trackStateContainer();
+      const auto& params = *fwdGsfResult.lastMeasurementState;
+      auto state =
+          m_propagator.template makeState<MultiComponentBoundTrackParameters,
+                                          decltype(bwdPropOptions),
+                                          MultiStepperSurfaceReached>(
```

Comment:
Same here?

Reply:
I think here it is not possible because we need to specialize on `MultiStepperSurfaceReached`